pybop/problems/fitting_problem.py

```diff
@@ -134,3 +135,112 @@ def evaluateS1(self, inputs: Inputs):
         )
 
         return (y, np.asarray(dy))
+
+
+class MultiFittingProblem(BaseProblem):
+    """
+    Problem class for joining mulitple fitting problems.
+
+    Extends `FittingProblem` to multiple fitting problems.
+    """
+
+    def __init__(self, problem_list, weights=None):
```

Comment:
I would covert this to unpack the args and form a list or dict with them. This is how we setup the `Parameters` class and provides a clean API :)

Something like:
```suggestion
    def __init__(self, *args, weights=None):
        self.problems = []
        for problem in args:
            self.problems.append(problem)
```